src/roiextractors/extractors/tiffimagingextractors/scanimagetiffimagingextractor.py

```diff
@@ -0,0 +1,102 @@
+"""Specialized extractor for reading TIFF files produced via ScanImage."""
+from pathlib import Path
+from typing import Optional, Tuple
+
+import numpy as np
+
+from ...extraction_tools import (
+    PathType,
+    check_get_frames_args,
+    FloatType,
+    ArrayType,
+)
+from ...imagingextractor import ImagingExtractor
+
+try:
+    from ScanImageTiffReader import ScanImageTiffReader
+
+    HAVE_SCAN_IMAGE_TIFF = True
+except ImportError:
+    HAVE_SCAN_IMAGE_TIFF = False
+
+
+class ScanImageTiffImagingExtractor(ImagingExtractor):
+    """Specialized extractor for reading TIFF files produced via ScanImage."""
+
+    extractor_name = "ScanImageTiffImaging"
+    installed = HAVE_SCAN_IMAGE_TIFF
+    is_writable = True
+    mode = "file"
+    installation_mesg = (
+        "To use the ScanImageTiffImagingExtractor install scanimage-tiff-reader: "
+        "\n\n pip install scanimage-tiff-reader\n\n"
+    )
+
+    def __init__(
+        self,
+        file_path: PathType,
+        sampling_frequency: FloatType,
+    ):
+        """
+        Specialized extractor for reading TIFF files produced via ScanImage.
+
+        The generic TiffImagingExtractor has issues loading these types of TIFF files in a lazy (memory mapped) manner.
+
+        The API calls to this extractor, however, have fully lazy behavior; however, direct slicing of the underlying
+        data structure is not equivalent to a numpy memory map.
+
+        Parameters
+        ----------
+        file_path : PathType
+            Path to the TIFF file.
+        sampling_frequency : float
+            The frequency at which the frames were sampled, in Hz.
+        """
+        assert self.installed, self.installation_mesg
+        super().__init__()
+        self.file_path = Path(file_path)
+        self._sampling_frequency = sampling_frequency
+        assert self.file_path.suffix in [".tiff", ".tif", ".TIFF", ".TIF"]
```

Comment:
I'd probably raise a warning here instead. It is possible that a tiff file might have a different suffix.

Reply:
Done